### PR TITLE
[WIP] added initial states to RNN and LSTM layers

### PR DESCRIPTION
**Pull request by fdlm**
This is a first implementation of initial states for RNN and LSTM layers. Needs some testing. Addresses #230.

### CODE REVIEW

File: madmom/ml/nn/layers.py [hunk 1]
```diff
@@ -283,16 +290,22 @@ class LSTMLayer(Layer):
...
     def __init__(self, input_gate, forget_gate, cell, output_gate,
-                 activation_fn=tanh):
+                 activation_fn=tanh, out_init=None, state_init=None):
         self.input_gate = input_gate
         self.forget_gate = forget_gate
         self.cell = cell
         self.output_gate = output_gate
         self.activation_fn = activation_fn
+        self.out_init = out_init
+        self.state_init = state_init
```
Comment (superbock) [inline madmom/ml/nn/layers.py]: As above, I'm not really happy with the names here.
Also, for streaming mode, the output of the previous step must be accessible anyways, e.g. as ``self.out``.

For the ``state_init`` I am thinking of moving this as ``init`` to the cell, and store the cell's state there -- this is basically what it is. This needs some change in the ``activate()`` method, but is the cleaner way to solve this (IMO).

**Comment by fdlm** [comment]
Since some of your proposed changes require more refactoring, I guess it would be really better to do this together with #185. It will be less work.

**Comment by superbock** [comment]
Yes, I will work on this next week, since we need NNs in #185 anyways.

**Comment by superbock** [comment]
Please see #235 for my first draft. If that PR addresses all your points we can close this PR.

**Comment by superbock** [comment]
Closing, since #243 is merged

**Pull request closed without merging**